interpreter.go

```diff
@@ -146,33 +176,203 @@ type valueSimpleObjectField struct {
 type valueSimpleObjectFieldMap map[string]valueSimpleObjectField
 
 type valueSimpleObject struct {
+	valueBase
 	upValues bindingFrame
 	fields   valueSimpleObjectFieldMap
 	asserts  []astNode
 }
 
+// findObject returns an object in which there is a given field.
+// It is used for field lookups, potentially involving super.
+func findObject(f string, curr value, startFrom int, counter *int) value {
+	switch curr := curr.(type) {
+	// case *valueExtendedObject:
+	// TODO(dcunnin): this
+
+	case *valueSimpleObject:
+		if *counter >= startFrom {
+			if _, ok := curr.fields[f]; ok {
+				return curr
+			}
+		}
+		*counter++
+
+		// case *valueComprehensionObject:
+		/*
+			if *counter >= startFrom {
+				// TODO(dcunnin): this
+			}
+			*counter++
+		*/
+	}
+	return nil
+}
+
+func objectIndex(loc *LocationRange, obj value, f string, offset int) (potentialValue, error) {
+	var foundAt int
+	found := findObject(f, obj, offset, &foundAt)
+	if found == nil {
+		return nil, makeRuntimeError(loc, fmt.Sprintf("Field does not exist: %s", f))
+	}
+	switch found := found.(type) {
+	case *valueSimpleObject:
+		self := found
+		offset := foundAt
+		upValues := bindingFrame{}
+		field := found.fields[f]
+
+		return makeThunk("???", makeEnvironment(upValues, self, offset), field.body), nil
```

Comment:
The evaluation of the body happens immediately, so no need to make a thunk, just recurse,